lib/modules/datasource/npm/yarnrc.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { loadConfigFromYarnrcYml, resolveRegistryUrl } from './yarnrc';
+
+describe('modules/datasource/npm/yarnrc', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('resolveRegistryUrl()', () => {
+    it('considers default registry', () => {
+      const registryUrl = resolveRegistryUrl('a-package', {
+        npmRegistryServer: 'https://private.example.com/npm',
+      });
+      expect(registryUrl).toBe('https://private.example.com/npm');
+    });
+
+    it('chooses matching scoped registry over default registry', () => {
+      const registryUrl = resolveRegistryUrl('@scope/a-package', {
+        npmRegistryServer: 'https://private.example.com/npm',
+        npmScopes: {
+          scope: {
+            npmRegistryServer: 'https://scope.example.com/npm',
+          },
+        },
+      });
+      expect(registryUrl).toBe('https://scope.example.com/npm');
+    });
+
+    it('ignores non matching scoped registry', () => {
+      const registryUrl = resolveRegistryUrl('@scope/a-package', {
+        npmScopes: {
+          'other-scope': {
+            npmRegistryServer: 'https://other-scope.example.com/npm',
+          },
+        },
+      });
+      expect(registryUrl).toBeNull();
+    });
+  });
+
+  describe('loadConfigFromYarnrcYml()', () => {
+    it.each([
+      [
+        'npmRegistryServer: https://npm.example.com',
+        { npmRegistryServer: 'https://npm.example.com' },
+      ],
+      [
+        `
```

Comment:
```suggestion
        codeBlock`
```

We should at least use  `codeBlock` helper then to get properly indented yaml

Reply:
Didn't know that - thanks!